mllib/src/main/scala/org/apache/spark/mllib/linalg/EigenValueDecomposition.scala

```diff
@@ -79,6 +79,10 @@ private[mllib] object EigenValueDecomposition {
     // Mode 1: A*x = lambda*x, A symmetric
     iparam(6) = 1
 
+    require(ncv * n.toLong <= Integer.MAX_VALUE, "Product of 2*k*n must be smaller than " +
```

Comment:
Sorry, one other tiny thing: the problematic product is not necessarily 2*k*n here. I think you could simplify these error messages to just say that the k and n are too large, rather than explain the computation. Right now the expressions are repeated 3 places. Might be nice to keep that to 2, and keep them together.